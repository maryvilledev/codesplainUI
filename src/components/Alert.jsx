import React, { PropTypes } from 'react';
import {
  Dialog,
  FlatButton,
} from 'material-ui';

/*
<Alert /> renders as a <Dialog /> with an OK button. It displays the text
specified in the 'text' prop, and invokes the 'onClose' prop when it is closed.
Setting the 'isError' prop to true causes its background to be rendered as a
crimson red.
*/
const Alert = ({ onClose, text }) => (
    <Dialog
      actions={
        <FlatButton
          label="OK"
          primary={true}
          onTouchTap={onClose}
        />
      }
      modal={false}
      open={true}
      onRequestClose={onClose}
    >
      {text}
    </Dialog>
);

Alert.propTypes = {
  onClose: PropTypes.func.isRequired,
  text: PropTypes.string.isRequired,
};

export default Alert;
