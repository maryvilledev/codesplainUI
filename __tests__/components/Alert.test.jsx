import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import getMuiTheme from 'material-ui/styles/getMuiTheme';

import Alert from '../../src/components/Alert';

describe('<Alert />', () => {
  const muiTheme = getMuiTheme();
  const shallowWithContext = (node) => shallow(node, { context: { muiTheme } });

  it('matches snapshot', () => {
    const wrapper = shallowWithContext(
      <Alert
        onClose={jest.fn()}
        text="Wubba lubba dub dub!"
      />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  describe('prop: onClose', () => {
    it('should forward it', () => {
      const onClose = jest.fn();
      const wrapper = shallowWithContext(
        <Alert
          onClose={onClose}
          text="Wubba lubba dub dub!"
        />
      );
      expect(wrapper.prop('onRequestClose')).toEqual(onClose);
    });
  });
});
