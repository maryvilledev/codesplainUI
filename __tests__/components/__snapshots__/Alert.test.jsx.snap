exports[`<Alert /> matches snapshot 1`] = `
<Dialog
  actions={
    <FlatButton
      disabled={false}
      fullWidth={false}
      label="OK"
      labelPosition="after"
      labelStyle={Object {}}
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      onTouchTap={[Function]}
      primary={true}
      secondary={false} />
  }
  autoDetectWindowHeight={true}
  autoScrollBodyContent={false}
  modal={false}
  onRequestClose={[Function]}
  open={true}
  repositionOnUpdate={true}>
  Wubba lubba dub dub!
</Dialog>
`;
